pkg/controller/volume/attachdetach/cache/desired_state_of_world.go

```diff
@@ -280,30 +284,42 @@ func (dsw *desiredStateOfWorld) DeletePod(
 	}
 
 	delete(
-		dsw.nodesManaged[nodeName].volumesToAttach[volumeName].scheduledPods,
+		dsw.nodesManaged[node].volumesToAttach[volumeName].scheduledPods,
 		podName)
 
 	if len(volumeObj.scheduledPods) == 0 {
 		delete(
-			dsw.nodesManaged[nodeName].volumesToAttach,
+			dsw.nodesManaged[node].volumesToAttach,
 			volumeName)
 	}
 }
 
-func (dsw *desiredStateOfWorld) NodeExists(nodeName k8stypes.NodeName) bool {
+func (dsw *desiredStateOfWorld) NodeExists(node k8stypes.NodeIdentifier) bool {
 	dsw.RLock()
 	defer dsw.RUnlock()
 
-	_, nodeExists := dsw.nodesManaged[nodeName]
+	_, nodeExists := dsw.nodesManaged[node]
 	return nodeExists
 }
 
+func (dsw *desiredStateOfWorld) FindNodeByName(nodeName k8stypes.NodeName) (k8stypes.NodeIdentifier, bool) {
+	dsw.RLock()
```

Comment:
If a node is restarted quickly, will this function actually find the wrong (old) node cached in dsw?